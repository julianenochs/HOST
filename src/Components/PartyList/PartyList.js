import React from 'react';
import { connect } from 'react-redux';
import { selectParty } from '../../Actions/index';
import './PartyList.scss';

export const PartyList = (props) => {
  return props.parties.map(party => {
    return(<ul>
        <button className='party__list'
            onClick={() => {
              props.selectParty(party.id)
              props.setCurrentParty(party.id)
            }}>{party.name}</button>
            </ul>
      )
    })
  }
  

const mapStateToProps = state => ({
  parties: state.parties,
});

const mapDispatchToProps = dispatch => ({
  selectParty: (partyId) => dispatch(selectParty(partyId))
})

export default connect(mapStateToProps, mapDispatchToProps)(PartyList);