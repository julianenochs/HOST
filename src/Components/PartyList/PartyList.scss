$primary-color: #e0c9b1;

.party__list {
  background-color: $primary-color;
  border: none;
  font-family: 'Federo', sans-serif;
  font-size: 2em;
    &:hover {
      text-shadow: 1px 1px 1px;
  }
}
