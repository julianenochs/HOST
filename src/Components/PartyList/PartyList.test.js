import { shallow } from 'enzyme';
import PartyList from './PartyList';
import React from 'react';

describe('Party List', () => {
  let wrapper;
  let mockSetCurrentParty = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<PartyList 
      setCurrentParty={mockSetCurrentParty}
      key={3}/>)
  });

  it.skip('should match the Party List Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});