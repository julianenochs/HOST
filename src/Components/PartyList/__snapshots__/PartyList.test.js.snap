// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Party List should match the Party List Snapshot 1`] = `undefined`;
